src/test/java/io/confluent/examples/streams/utils/PairSerializer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.confluent.examples.streams.utils;
+
+import org.apache.kafka.common.serialization.Serializer;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Map;
+
+public class PairSerializer<X, Y> implements Serializer<Pair<X, Y>> {
+
+  private final Serializer<X> serializerX;
+  private final Serializer<Y> serializerY;
+
+  public PairSerializer(final Serializer<X> serializerX, final Serializer<Y> serializerY) {
+    this.serializerX = serializerX;
+    this.serializerY = serializerY;
+  }
+
+  @Override
+  public void configure(final Map<String, ?> configs, final boolean isKey) {
+  }
+
+  @Override
+  public byte[] serialize(final String topic, final Pair<X, Y> pair) {
+    if (pair == null) {
+      return null;
+    }
+    final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    try {
+      final DataOutputStream out = new DataOutputStream(baos);
+      final byte[] serializedX = serializerX.serialize(topic, pair.x);
+      final byte[] serializedY = serializerY.serialize(topic, pair.y);
+      short magicByte = 0;
+      if (serializedX != null) {
+        magicByte += 1;
+      }
+      if (serializedY != null) {
+        magicByte += 2;
+      }
+      out.writeShort(magicByte);
+      if (magicByte == 1) {
+        writeX(out, serializedX);
+      }
+      if (magicByte == 2) {
+        writeY(out, serializedY);
+      }
+      if (magicByte == 3) {
+        writeX(out, serializedX);
+        writeY(out, serializedY);
+      }
+      out.close();
```

Comment:
move to `finally` block or better use try-with-resource construct.

Reply:
Ack, using try-with-resources.